### DISCUSSION TITLE
Support for CloudWatch Metrics unit specification

### DISCUSSION THREAD

**Discussion by robbytx** [vectordotdev/vector#21763] [General]
CloudWatch metrics supports the specification of the [unit of measurement](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Unit), like Percent, Bytes, Megabytes, Count, etc, which permits the graphs to be labelled and even scaled appropriately to aid in interpretation.

Is there (or could there be) some way to have the [CloudWatch Metrics sink](https://vector.dev/docs/reference/configuration/sinks/aws_cloudwatch_metrics/) use a property/tag on the metric value to pass as the [Unit](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html) property in its PutMetricData API call?

**Comment by jszwedko** [comment]
I think the most robust approach would be add a field to Vector's metric data model that would be the `unit`. I think we can take inspiration from OTLP's model. This would be a broad change though.
